service/controller/app/resource/chartoperator/resource.go

```diff
@@ -140,6 +149,79 @@ func (r Resource) installChartOperator(ctx context.Context, cr v1alpha1.App) err
 	return nil
 }
 
+func (r Resource) triggerReconciliation(ctx context.Context, operatorApp v1alpha1.App) error {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// Find all the Apps CR for a given cluster.
+	// If `operatorApp` is an org-namespaced App use the cluster label selector.
+	var appList v1alpha1.AppList
+	{
+		o := client.ListOptions{}
+
+		var selector k8slabels.Selector
+
+		if key.IsInOrgNamespace(operatorApp) {
+			selector, err = k8slabels.Parse(fmt.Sprintf("%s=%s", label.Cluster, key.ClusterLabel(operatorApp)))
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			o.LabelSelector = selector
+		}
+
+		err = r.ctrlClient.List(ctx, &appList, &o)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	// For each App, check if the corresponding Chart CR exists.
+	// If not, annotate the App to trigger the reconciliation.
+	for i, app := range appList.Items {
+		// Skip for in-cluster apps and the chart-operator app itself.
+		if key.InCluster(app) || app.ObjectMeta.Name == operatorApp.ObjectMeta.Name {
```

Comment:
Why skip for incluster apps?

Reply:
Well, the honest answer is that I thought this logic matters only for bootstrapping WC and its apps, and then I also thought all of them are in-cluster apps, except for the app-operator 🤔

Reply:
Yes, this is just for speeding up the install of workload apps in new clusters. The MC apps in the collection are reconciled by flux and the new types in-cluster apps aren't affected.